front/pages/api/poke/workspaces/[wId]/data_sources/[name]/bot_enabled.ts

```diff
@@ -0,0 +1,154 @@
+      api_error: {
+        type: "user_not_found",
+        message: "Could not find the user.",
+      },
+    });
+  }
+
+  if (!user.isDustSuperUser) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "user_not_found",
+        message: "Could not find the user.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "POST":
+      const { wId } = req.query;
+      if (!wId || typeof wId !== "string") {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message:
+              "The request query is invalid, expects { workspaceId: string }.",
+          },
+        });
+      }
+
+      const { name } = req.query;
+      if (name !== "managed-slack") {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message:
+              "The request query is invalid, expects name to be managed-slack.",
+          },
+        });
+      }
+
+      if (!req.body || typeof req.body.botEnabled !== "boolean") {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message:
+              "Missing required parameters. Required: botEnabled (boolean)",
+          },
+        });
+      }
+      const { botEnabled } = req.body;
+
+      const workspace = await Workspace.findOne({
+        where: {
+          sId: wId,
+        },
+      });
+
+      if (!workspace) {
+        return apiError(req, res, {
+          status_code: 404,
+          api_error: {
+            type: "workspace_not_found",
+            message: "Could not find the workspace.",
+          },
+        });
+      }
+
+      const dataSource = await DataSource.findOne({
+        where: {
+          workspaceId: workspace.id,
+          name,
+        },
+      });
+
+      if (!dataSource) {
+        return apiError(req, res, {
+          status_code: 404,
+          api_error: {
+            type: "data_source_not_found",
+            message: "Could not find the data source.",
+          },
+        });
+      }
+
+      if (!dataSource.connectorId) {
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "internal_server_error",
+            message: "ConnectorId not set.",
+          },
+        });
+      }
+
+      const connectorRes = await ConnectorsAPI.setBotEnabled(
+        dataSource.connectorId,
+        botEnabled
+      );
+
+      if (connectorRes.isErr()) {
+        return apiError(
+          req,
+          res,
+          {
+            status_code: 500,
+            api_error: {
+              type: "internal_server_error",
+              message: `An error occurred while enabling the bot`,
+            },
+          },
+          new Error(JSON.stringify(connectorRes.error.error))
```

Comment:
It's not in the current code, it's a ConnectorsAPIResponse which is in turn a Result<T, ConnectorsAPIErrorResponse> (from the generic _resultFromResponse of connectors_api.ts)

I am hesitant to change it since it's the way it's done throughout all the rest of this part of the code. Do you have a preference?

Reply:
ok, done. Ready to merge if no other com